blablacar-service/src/main/java/cz/fi/muni/pa165/teamred/service/facade/UserFacadeImpl.java

```diff
@@ -48,48 +49,105 @@ public Long createUser(UserCreateDTO user){
 
     @Override
     public void editUser(UserDTO userDTO) {
-        throw new NotImplementedException();
+        User user = new User();
+        user.setId(userDTO.getId());
+        user.setName(userDTO.getName());
+        user.setSurname(userDTO.getSurname());
+        user.setNickname(userDTO.getNickname());
+        userService.editUser(user);
     }
 
     @Override
     public void removeUser(Long userId) {
-        throw new NotImplementedException();
+        //redundant object creation ?
+        User userToDelete = new User();
+        userToDelete.setId(userId);
+        userService.deleteUser(userToDelete);
     }
 
-    @Override
-    public void addComment(Long userId, Long commentId) {
-        userService.addUserComment(userService.findUserById(userId),
-                commentService.findById(commentId));
-        log.info("Added Comment: " + commentService.findById(commentId).toString() +
-                "to User: " + userService.findUserById(userId).toString());
-    }
-
-    @Override
-    public void addRideAsDriver(Long userId, Long rideId) {
-        throw new NotImplementedException();
-    }
 
     @Override
     public void addRideAsPassenger(Long userId, Long rideId) {
-        throw new NotImplementedException();
+        User user = userService.findUserById(userId);
+        if (user == null){
+            log.info("No user found with id: " + userId);
+            return;
+        }
+        Ride ride = rideService.findById(rideId);
+        if (ride == null){
+            log.info("No ride found with id: " + rideId);
+            return;
+        }
+
+        userService.addUserRideAsPassenger(user, ride);
+        rideService.addPassenger(ride,user);
+        log.info("Added user: " + user +
+        " as a passenger in ride: " + ride);
     }
 
     @Override
     public void removeComment(Long userId, Long commentId) {
-        userService.removeUserComment(userService.findUserById(userId),
-                commentService.findById(commentId));
-        log.info("Removed Comment: " + commentService.findById(commentId).toString() +
-        "from User: " + userService.findUserById(userId).toString());
+        User user = userService.findUserById(userId);
+        if (user == null){
+            log.info("No user found with id: " + userId);
+            return;
+        }
+        Comment comment = commentService.findById(commentId);
+        if (comment == null){
+            log.info("No comment found with id: " + commentId);
+            return;
+        }
+
+        //deleting both sides of references
+        userService.removeUserComment(user,comment);
+        log.info("Removed Comment: " + comment.toString() +
+        " and comment reference from User: " + user.toString());
+
+        commentService.deleteComment(comment);
     }
 
     @Override
     public void removeRideAsDriver(Long userId, Long rideId) {
```

Comment:
Logika sedi, ked nechce user uz byt sofer tak sa vymaze ride. Ked si ale predstavim frontend tak user si bude prehliadat svoje rides kde je driver a bude mat nejaky button "Delete Ride". Tu bude mat vacsiu logiku ked frontend bude do api volat delete na rides a tym padom sa mi to viac hodi do RideFacade

Reply:
jo to uz som davno vymazal take ako vidim je to stary PR